backend/lib/models/schemas/v2/post.js

```diff
@@ -0,0 +1,169 @@
+// -- Imports
+const { Schema, model, ObjectId } = require("mongoose");
+const { schema: authorSchema } = require("./author");
+
+// -- Schema
+const postSchema = new Schema(
+  {
+    author: Object,
+    content: {
+      required: true,
+      trim: true,
+      type: String,
+    },
+    expireAt: Date,
+    externalLinks: {
+      appStore: String,
+      email: String,
+      playStore: String,
+      website: String,
+    },
+    language: [String],
+    likes: {
+      // TODO: how to guarantee unique ids?
```

Comment:
Maybe we can use `enum` to check if ids exist already or not?

Reply:
Actually, I've used `addToSet` and `removeToSet` mongo operations at the endpoints for liking, so that will guarantee unicity at DB level, though not validate at the model. With that, if the ID is already there, it won't return an error or a message, and neither will it add the like again.

How could we use `enum` to validate at the model level?

Reply:
Yea adding a validation at model-level sounds like a good idea for error handling and potentially catching ids that pass the model validation but are not unique.

Now that I think of it `enum` won't work because I believe you need to define the values first, which doesn't sound right for likes. 

Instead, could we use unique indexes? If I remember how it works, we could add an index on both the `userId` and `postId` to prevent duplicate insertions.

Reply:
@Naraujo13 sounds good, first I used that `PUT likes/:id` in the API because I found it somewhere and thought it makes sense in a RESTful way, but I think what I did wat return a 400 bad request if the like already existed, which isn't completely valid in REST I think since `PUT` is idempotent so it always has the same result, whether or not the like already existed, after the request it's certainly `true`. 
For DELETE this is similar, after the DELETE the resources is certainly deleted, but I think it makes sense to return a `404` error as well if it was already deleted at the time of the request.

Reply:
@anniegiang I don't think this is needed per se to add at model level as unique index because this is not a separate collection I think, it's just an array inside a document... and the mongo query as I implemented it before would just not add duplicates so that should be ok... not sure if that changed now?

Reply:
It is possible to create a unique index, but that will guarantee only at db level, the same way we are already doing with `addToSet` for example.

@robinv85 currently both `PUT` and `DELETE` are idempotent, though no error if the like is already there or if it has already been deleted.

Reply:
Yeah they both should be idempotent but I'm not sure if the response has to follow a certain rule, we might be open to decide this for our own, but it would be nice to have a consistent approach accros all API handlers. 
Also for the client, I don't think this makes a big difference, if we always return OK then the client will just update the state according to the request too. Iimagine 2 people click delete at same time, both would get "ok" and no error message, and client would just remove the post, so seems good to me.